node/src/actors/chain_manager/mod.rs

```diff
@@ -691,6 +702,74 @@ impl ChainManager {
             .clone()
     }
 
+    fn add_temp_superblock_votes(&mut self, ctx: &mut Context<Self>) -> Result<(), failure::Error> {
+        for superblock_vote in std::mem::take(&mut self.temp_superblock_votes) {
+            log::debug!("add_temp_superblock_votes {:?}", superblock_vote);
+            // Check if we already received this vote
+            if self.chain_state.superblock_state.contains(&superblock_vote) {
+                return Ok(());
+            }
+
+            // Validate secp256k1 signature
+            signature_mngr::verify_signatures(vec![SignaturesToVerify::SuperBlockVote {
+                superblock_vote: superblock_vote.clone(),
+            }])
+            .map_err(|e| {
+                log::error!("Verify superblock vote signature: {}", e);
+            })
+            .into_actor(self)
+            .and_then(move |(), act, _ctx| {
+                // Check if we already received this vote (again, because this future can be executed
+                // by multiple tasks concurrently)
+                if act.chain_state.superblock_state.contains(&superblock_vote) {
+                    return actix::fut::ok(());
+                }
+                // Validate vote: the identity should be able to vote
+                // We broadcast all superblock votes with valid secp256k1 signature, signed by members
+                // of the ARS, even if the superblock hash is different from our local superblock hash.
+                // If the superblock index is different from the current one we cannot check ARS membership,
+                // so we broadcast it if the index is within an acceptable range (not too old).
+                let _should_broadcast =
+                    match act.chain_state.superblock_state.add_vote(&superblock_vote) {
```

Comment:
_should_broadcast is not used, it looks like a copy past from other part, but it is not clear to me why it is used here more than add_vote and show a log message

Reply:
Due to there are 2 copies of that part of code, should we create a function?

Reply:
Nice one!